dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/database/migration/V9_02_02__DML_Fill_urlaliases.java

```diff
@@ -0,0 +1,285 @@
+    ENTITYMIGRATIONTABLES.put(AUDIO, null);
+    ENTITYMIGRATIONTABLES.put(BOOK, null);
+    ENTITYMIGRATIONTABLES.put(COLLECTION, "collections");
+    ENTITYMIGRATIONTABLES.put(CORPORATE_BODY, "corporatebodies");
+    // ENTITYMIGRATIONTABLES.put(DIGITAL_OBJECT, "digitalobjects");
+    ENTITYMIGRATIONTABLES.put(ENTITY, null);
+    ENTITYMIGRATIONTABLES.put(EVENT, null);
+    ENTITYMIGRATIONTABLES.put(EXPRESSION, null);
+    ENTITYMIGRATIONTABLES.put(FAMILY, "familynames");
+    ENTITYMIGRATIONTABLES.put(GEOLOCATION, "geolocations");
+    ENTITYMIGRATIONTABLES.put(HEADWORD_ENTRY, null);
+    ENTITYMIGRATIONTABLES.put(IMAGE, null);
+    ENTITYMIGRATIONTABLES.put(ITEM, "items");
+    ENTITYMIGRATIONTABLES.put(MANIFESTATION, null);
+    ENTITYMIGRATIONTABLES.put(OBJECT_3D, null);
+    ENTITYMIGRATIONTABLES.put(PERSON, "persons");
+    ENTITYMIGRATIONTABLES.put(PLACE, null);
+    ENTITYMIGRATIONTABLES.put(PROJECT, "projects");
+    ENTITYMIGRATIONTABLES.put(TOPIC, "topics");
+    ENTITYMIGRATIONTABLES.put(VIDEO, null);
+    ENTITYMIGRATIONTABLES.put(WEBSITE, "websites");
+    ENTITYMIGRATIONTABLES.put(WORK, "works");
+  }
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(V9_02_02__DML_Fill_urlaliases.class);
+
+  private SlugGenerator slugGenerator = new SlugGenerator();
+
+  @Override
+  public void migrate(Context context) throws Exception {
+
+    final SingleConnectionDataSource connectionDataSource =
+        new SingleConnectionDataSource(context.getConnection(), true);
+    JdbcTemplate jdbcTemplate = new JdbcTemplate(connectionDataSource.getConnection());
+
+    // Webpages have to be migrated individually, since their URLAliases depend on the
+    // the websites, they are bound, too
+    migrateWebpages(jdbcTemplate);
+
+    // All other entities have only generic URLAliases
+    ENTITYMIGRATIONTABLES.forEach(
+        (entityType, tableName) -> {
+          if (tableName != null) {
+            try {
+              migrateIdentifiables(jdbcTemplate, tableName, entityType);
+            } catch (SQLException e) {
+              throw new RuntimeException("Cannot migrate " + entityType + ": " + e, e);
+            }
+          }
+        });
+
+    // FIXME Das ist nur zu Demonstrationszwecken!
+    throw new RuntimeException("Force trigger rollback");
+  }
+
+  private void migrateWebpages(JdbcTemplate jdbcTemplate) throws SQLException {
+    String selectQuery =
+        "SELECT w.uuid AS w_uuid, w.label AS label, ww.website_uuid AS ws_uuid FROM webpages w, website_webpages ww WHERE ww.webpage_uuid=w.uuid ORDER BY ws_uuid,w_uuid";
+    List<Map<String, String>> webpages = jdbcTemplate.queryForList(selectQuery);
+    webpages.forEach(
+        w -> {
+          JSONObject jsonObject = new JSONObject(w.toString());
+          UUID uuid = UUID.fromString(jsonObject.getString("w_uuid"));
+          UUID websiteUuid = UUID.fromString(jsonObject.getString("ws_uuid"));
+          try {
+            HashMap labels =
```

Comment:
```suggestion
            Map<String, String> labels =
```
dann könnten wir unten die casts weglassen...